webrtc/videoroom/assets/src/membraneWebRTC.ts

```diff
@@ -48,40 +57,51 @@ interface Callbacks {
   onDisplayStream?: (stream: MediaStream, label: string) => void;
   onDisplayTrack?: (ctx: TrackContext) => void;
   onHideTrack?: (ctx: TrackContext) => void;
-  onParticipantToggledVideo?: (streamId: string) => void;
-  onParticipantToggledAudio?: (streamId: string) => void;
+  onParticipantToggledVideo?: (participantId: string) => void;
+  onParticipantToggledAudio?: (participantId: string) => void;
   onOfferData?: (data: OfferData) => void;
+  onNoMediaParticipantArrival?: (participant: Participant) => void;
+  onNoMediaParticipantLeave?: (participant: Participant) => void;
+  onParticipantsList?: (participants: Participant[]) => void;
 }
 
 interface MembraneWebRTCConfig {
   callbacks?: Callbacks;
   rtcConfig?: RTCConfiguration;
   type?: "participant" | "screensharing";
-  displayName: string;
+  signalingOptions: SignalingOptions;
 }
 
 export class MembraneWebRTC {
   private readonly socket: Socket;
   private _channel?: Channel;
   private channelId: string;
   private socketRefs: string[] = [];
+  private signalingOptions: SignalingOptions;
   private displayName: string;
 
   private maxDisplayNum: number = 1;
-  private localTracks: Set<MediaStreamTrack> = new Set<MediaStreamTrack>();
-  private localStream?: MediaStream;
+  private localTracksMapping: Map<
+    String,
+    [MediaStreamTrack, MediaStream]
+  > = new Map();
   private screensharingStream?: MediaStream;
   private remoteStreams: Set<MediaStream> = new Set<MediaStream>();
   private midToStream: Map<String, MediaStream> = new Map();
   private connection?: RTCPeerConnection;
   private participants: Participant[] = [];
+  private midToParticipant: Map<String, Participant> = new Map<
+    String,
+    Participant
+  >();
   private readonly rtcConfig: RTCConfiguration = {
     iceServers: [
       {
         urls: "stun:stun.l.google.com:19302",
       },
     ],
   };
+  private userId?: string;
```

Comment:
`userId` is equal to value of field `id` in `Participant` representing the recipient of the message

Reply:
I am not sure why we have userId nad Prticipant.id. IMO we should have only one id that should be assigned to participant when he joins to the room i.e. here: https://github.com/membraneframework/membrane_demo/blob/master/webrtc/videoroom/lib/videoroom/pipeline.ex#L95

Can't we also use Endpoint.id instead of creating a new UUID.v4()?